### PR TITLE
Factored out coverage map from the Progress tracker 

### PR DESCRIPTION
**Pull request by nflearl**
**JIRA Tickets:**

[AB2D-3824](https://jira.cms.gov/browse/AB2D-3824) - Factor out coverage map.

***Related Tickets***
 
### What Does This PR Do?

Internal refactoring of coverage map as it really does not belong in the tracker.  This is a foundational step towards a more streaming model.

### What Should Reviewers Watch For?
There should be no differences in behavior.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [x] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ x] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java [hunk 1]
```diff
@@ -114,9 +113,7 @@ public ContractProcessorImpl(JobRepository jobRepository,
 
             contractData.setStreamHelper(helper);
 
-            Iterator<Map.Entry<String, CoverageSummary>> patientEntries = patients.entrySet().iterator();
-
-            while (patientEntries.hasNext()) {
+            for (Map.Entry<String, CoverageSummary> stringCoverageSummaryEntry : patients.entrySet()) {
```
Comment (sb-wnyffenegger) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java:116]: Is this the best variable name? Each entry is a coverage summary for a beneficiary.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java [hunk 2]
```diff
@@ -231,6 +235,7 @@ private Path createOutputDirectory(Path outputDirPath, Job job) {
     private void deleteExistingDirectory(Path outputDirPath, Job job) {
         final File[] files = outputDirPath.toFile().listFiles(getFilenameFilter());
 
+        assert files != null;
```
Comment (sb-wnyffenegger) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java:238]: Wouldn't it be better to throw a runtime exception? Quartz catches those and logs them in a standard way.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ProgressTracker.java [hunk 3]
```diff
@@ -73,7 +63,7 @@ public void addPatients(List<CoverageSummary> coverage) {
      * @return number of patients
      */
     public int getTotalCount() {
-        return patients.size();
+        return metadataProcessedCount;
```
Comment (sb-wnyffenegger) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/ProgressTracker.java:66]: Why not expected beneficiaries?

**Review by sb-wnyffenegger — COMMENTED** [review]
Relatively minor comments but tests are failing

File: worker/src/main/java/gov/cms/ab2d/worker/processor/ProgressTracker.java (hunk 3 above)
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/ProgressTracker.java:66]: Ok, wasn't quite sure the best replacement.  Fixed.

**Review by sharshar-sb — APPROVED** [review]

**Review by sb-wnyffenegger — APPROVED** [review]

**Review by sb-wnyffenegger — APPROVED** [review]

**Review by sharshar-sb — APPROVED** [review]

**Pull request merged**